### PR TITLE
[Randomness][Fix] check feature flags

### PR DESCRIPTION
**Pull request by danielxiangzl**
### Description
Check if vtxn feature is enabled for DKG and randomness besides checking DKG.
Check if DKG feature is enabled in consensus besides checking vtxn.

### CODE REVIEW

File: consensus/src/counters.rs [hunk 1]
```diff
@@ -843,6 +843,15 @@ pub static UNEXPECTED_PROPOSAL_EXT_COUNT: Lazy<IntCounter> = Lazy::new(|| {
...
 });
 
+/// Count of the number of `DKG` validator transactions received while the feature is disabled.
+pub static UNEXPECTED_DKG_VTXN_COUNT: Lazy<IntCounter> = Lazy::new(|| {
+    register_int_counter!(
+        "aptos_consensus_unexpected_dkg_vtxn_count",
+        "Count of the number of `DKG` validator transactions received while the feature is disabled."
+    )
+    .unwrap()
+});
```
Comment (zjma) [inline consensus/src/counters.rs:853]: how do we want to consume this metric?

Comment (danielxiangzl) [reply consensus/src/counters.rs:853]: Same as `UNEXPECTED_PROPOSAL_EXT_COUNT`?

File: types/src/validator_txn.rs [hunk 2]
```diff
@@ -40,6 +40,14 @@ impl ValidatorTransaction {
     pub fn size_in_bytes(&self) -> usize {
         bcs::serialized_size(self).unwrap()
     }
+
+    pub fn topic(&self) -> Topic {
+        match self {
+            Self::DummyTopic1(_) => Topic::DUMMY1,
+            Self::DKGResult(_) => Topic::DKG,
+            Self::DummyTopic2(_) => Topic::DUMMY1,
```
Comment (zjma) [inline types/src/validator_txn.rs:48]:
```suggestion
            Self::DummyTopic2(_) => Topic::DUMMY2,
```

File: consensus/src/counters.rs (hunk 1 above)
Comment (zjma) [reply consensus/src/counters.rs:853]: still not sure how :)

File: consensus/src/epoch_manager.rs [hunk 3]
```diff
@@ -1115,7 +1121,11 @@ impl<P: OnChainConfigProvider> EpochManager<P> {
         new_epoch_state: &EpochState,
         features: &Features,
         maybe_dkg_state: anyhow::Result<DKGState>,
+        consensus_config: &OnChainConsensusConfig,
     ) -> Result<RandConfig, NoRandomnessReason> {
+        if !consensus_config.is_vtxn_enabled() {
+            return Err(NoRandomnessReason::VTxnDisabled);
+        }
```
Comment (zjma) [inline consensus/src/epoch_manager.rs:1128]: I'm not sure if this check is necessary.

File: consensus/src/counters.rs (hunk 1 above)
Comment (danielxiangzl) [reply consensus/src/counters.rs:853]: Will have a figure showing error rate, similar to [this](https://aptoslabs.grafana.net/d/consensus/consensus?from=now-1h&to=now&var-Datasource=fHo-R604z&var-BigQuery=axNEitxVz&var-metrics_source=vmagent&var-chain_name=randomnet&var-cluster=All&var-namespace=All&var-kubernetes_pod_name=All&orgId=1&editPanel=140)

**Review by zjma — APPROVED** [review]

File: consensus/src/epoch_manager.rs (hunk 3 above)
Comment (danielxiangzl) [reply consensus/src/epoch_manager.rs:1128]: We should only start rand manager when vtxn is enabled? Do you see any downside?

Comment (zjma) [reply consensus/src/epoch_manager.rs:1128]: no downside too 😅

**Pull request merged**